urnx/doc.go

```diff
@@ -0,0 +1,16 @@
+// Copyright 2022 The Infratographer Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package urnx is a package for creating and parsing Infratographer based URNs
```

Comment:
A description of what an Infratographer URN is (purpose, format, etc) would be good here.

It looks like we're also opting for `:` as our separator between resource type and identifier for the NSS section of the URN. That's fine - it's compatible with older URN RFCs - but [RFC 8141][rfc-8141] does allow for `/` as a character in URN NSSes as well. Just food for thought - not saying we need to change anything.

[rfc-8141]: https://www.rfc-editor.org/rfc/rfc8141.html#section-2

Reply:
✅